nodes/nomos-node-api/src/lib.rs

```diff
@@ -53,13 +55,16 @@ where
     /// Initialize the service with the given state
     fn init(service_state: ServiceStateHandle<Self>) -> Result<Self, DynError> {
         let settings = service_state.settings_reader.get_updated_settings();
-        Ok(Self { settings })
+        Ok(Self {
+            settings,
+            handle: service_state.overwatch_handle,
+        })
     }
 
     /// Service main loop
     async fn run(mut self) -> Result<(), DynError> {
         let endpoint = B::new(self.settings.backend_settings).await?;
-        endpoint.serve().await?;
+        endpoint.serve(self.handle).await?;
         Ok(())
     }
 }
```

Comment:
This probably needs to be updated. There are things to do here. Usually you would want the service to hold the whole service state.  Lifecycle is ready as well. **Not to be done now**, but lets open an issue to handle the service here. I can probably implement that myself.

Reply:
Created at https://github.com/logos-co/nomos-node/issues/500.